### ISSUE TITLE
Game crashes on Tomahawk missile fire

### ISSUE THREAD

**Issue by dmzorev** [Inq8/CAmod#57]
The game sometimes crashes when Tomahawk launches missiles.

```
Exception of type `System.InvalidOperationException`: Attempting to query the position of an invalid Target
   at OpenRA.Traits.Target.get_CenterPosition() in /<...>/CAmod/engine/OpenRA.Game/Traits/Target.cs:line 169
   at OpenRA.Mods.CA.Activities.CruiseMissileFly..ctor(Actor self, Target t, CruiseMissile cm, WDist maxAltitude, WDist maxTargetMovement, Boolean trackTarget) in <...>/CAmod/OpenRA.Mods.CA/Activities/CruiseMissileFly.cs:line 36
   at OpenRA.Mods.CA.Traits.CruiseMissile.GetActivity(Actor self, Target target) in /<...>/CAmod/OpenRA.Mods.CA/Traits/CruiseMissile.cs:line 91
   at OpenRA.Mods.CA.Traits.MissileBase.OpenRA.Traits.INotifyAddedToWorld.AddedToWorld(Actor self) in /<...>/CAmod/OpenRA.Mods.CA/Traits/MissileBase.cs:line 129
   at OpenRA.World.Add(Actor a) in <...>/CAmod/engine/OpenRA.Game/World.cs:line 341
   at OpenRA.World.Tick() in <...>/CAmod/engine/OpenRA.Game/World.cs:line 453
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in /<...>/CAmod/engine/OpenRA.Game/Game.cs:line 630
   at OpenRA.Game.LogicTick() in <...>/CAmod/engine/OpenRA.Game/Game.cs:line 645
   at OpenRA.Game.Loop() in <...>/CAmod/engine/OpenRA.Game/Game.cs:line 812
   at OpenRA.Game.Run() in <...>/CAmod/engine/OpenRA.Game/Game.cs:line 862
   at OpenRA.Game.InitializeAndRun(String[] args) in <...>/CAmod/engine/OpenRA.Game/Game.cs:line 304
   at OpenRA.Launcher.Program.Main(String[] args) in <...>/CAmod/engine/OpenRA.Launcher/Program.cs:line 32

```

Game builded from `WIP` branch with new GDI tomahawk.

**Comment by dzorev** [comment]
Maybe problem in `OpenRA.Mods.CA.Activities.CruiseMissileFly.SetTarget(Target target)` method. 

Any way - needed some checks for `TargetType.Invalid`

**Comment by dmzorev** [comment]
Maybe need checks `OpenRA.Traits.Target.IsValidFor(Actor targeter)` before `SetTarget(Target target)` to missiles

**Comment by darkademic** [comment]
> Maybe need checks `OpenRA.Traits.Target.IsValidFor(Actor targeter)` before `SetTarget(Target target)` to missiles

I think this was fixed on 7th Jan (merged into WIP on 14th), have you pulled changes since then?

**Comment by dmzorev** [comment]
Also has another crash today

Builded from latest commit:

Commit name: Made Battlefortress French-ish
Revision Number: 72dfd56eaaed7ab5112d9e0a7a529757c7da9f53
Date: 28.01.2023

**Comment by darkademic** [comment]
Fixed added in 1.02-DevTest-3.
